File: qc-domain/src/main/java/com/eager/questioncloud/domain/creator/CreatorStatisticsRepository.java
package com.eager.questioncloud.domain.creator;

import java.util.List;

public interface CreatorStatisticsRepository {
    void save(CreatorStatistics creatorStatistics);

    void saveAll(List<CreatorStatistics> creatorStatistics);

    CreatorStatistics findByCreatorId(Long creatorId);
}

File: qc-core/qc-creator/qc-creator-core/src/main/kotlin/com/eager/questioncloud/creator/repository/CreatorStatisticsRepository.kt
package com.eager.questioncloud.creator.repository

import com.eager.questioncloud.creator.domain.CreatorStatistics

interface CreatorStatisticsRepository {
    fun save(creatorStatistics: CreatorStatistics)
    
    fun update(creatorStatistics: CreatorStatistics)
    
    fun findByCreatorId(creatorId: Long): CreatorStatistics
    
    fun getForUpdate(creatorId: Long): CreatorStatistics
    
    fun findByCreatorIdIn(creatorIds: List<Long>): Map<Long, CreatorStatistics>
    
    fun addSalesCount(creatorId: Long, count: Int)
    
    fun deleteAllInBatch()
}
